modular_skyrat/modules/modular_items/lewd_items/code/lewd_chemistry/chemistry_for_ERP.dm

```diff
@@ -1,181 +1,209 @@
+// The base ERP chem. It handles pref and human type checks for you, so ALL chems relating to ERP should be subtypes of this.
+/datum/reagent/drug/aphrodisiac
+	name = "liquid ERP"
+	description = "ERP in its liquified form. Complain to a coder."
 
-//Crocin. Basic aphrodisiac with no consequences
+	/// What preference you need enabled for effects on life
+	var/life_pref_datum = /datum/preference/toggle/erp
+	/// What preference you need enabled for effects on overdose
+	var/overdose_pref_datum = /datum/preference/toggle/erp
+
+/// Runs on life after preference checks. Use this instead of on_mob_life
+/datum/reagent/drug/aphrodisiac/proc/life_effects(mob/living/carbon/human/exposed_mob)
+	return
+
+/// Runs on OD process after preference checks. Use this instead of overdose_process
+/datum/reagent/drug/aphrodisiac/proc/overdose_effects(mob/living/carbon/human/exposed_mob)
+	return
+
+/datum/reagent/drug/aphrodisiac/on_mob_life(mob/living/carbon/human/exposed_mob)
+	. = ..()
+	if(life_pref_datum && exposed_mob.client?.prefs.read_preference(life_pref_datum) && ishuman(exposed_mob))
+		life_effects(exposed_mob)
+
+/datum/reagent/drug/aphrodisiac/overdose_process(mob/living/carbon/human/exposed_mob)
+	. = ..()
+	if(overdose_pref_datum && exposed_mob.client?.prefs.read_preference(overdose_pref_datum) && ishuman(exposed_mob))
+		overdose_effects(exposed_mob)
+
+/*
+* APHRODISIACS
+*/
 
-/datum/reagent/drug/crocin
-	name = "Crocin"
+//Crocin. Basic aphrodisiac with no consequences
+/datum/reagent/drug/aphrodisiac/crocin
+	name = "crocin"
 	description = "Naturally found in the crocus and gardenia flowers, this drug acts as a natural and safe aphrodisiac."
 	taste_description = "strawberries"
-	color = "#FFADFF"//PINK, rgb(255, 173, 255)
-	//can_synth = FALSE
-
-/datum/reagent/drug/crocin/on_mob_life(mob/living/carbon/human/M)
-	if(M.client && (M.client.prefs.read_preference(/datum/preference/toggle/erp/aphro)))
-		if((prob(min(current_cycle/2,5))))
-			M.emote(pick("moan","blush"))
-		if(prob(min(current_cycle/4,10)))
-			var/aroused_message = pick("You feel frisky.", "You're having trouble suppressing your urges.", "You feel in the mood.")
-			to_chat(M, span_notice("[aroused_message]"))
-		if(ishuman(M))
-			M.adjustArousal(1)
-			for(var/obj/item/organ/genital/G in M.internal_organs)
-				if(!G.aroused == AROUSAL_CANT)
-					G.aroused = AROUSAL_FULL
-					G.update_sprite_suffix()
-			M.update_body()
-	..()
+	color = "#FFADFF"
+	life_pref_datum = /datum/preference/toggle/erp/aphro
+
+	//Probabilities of flavour messages, as a %, run on mob life
+	var/emote_probability = 3
+	var/thought_probability = 2
+
+	//The possible contents of those flavour messages
+	var/list/possible_aroused_emotes = list("moan", "blush")
+	var/list/possible_aroused_thoughts = list("You feel frisky.", "You're having trouble suppressing your urges.", "You feel in the mood.")
+
+	var/arousal_adjust_amount = 1
+	var/pleasure_adjust_amount = 0
+	var/pain_adjust_amount = 0
```

Comment:
I think we should properly auto-doc this too.

Reply:
I felt the variables were fairly self explanatory, but I can certainly auto-doc them regardless